func_sig_registry/utils/abi.py

```diff
@@ -135,7 +136,36 @@ def event_definition_to_text_signature(abi: Dict[str, Any]) -> str:
     return '{fn_name}({fn_input_types})'.format(
         fn_name=abi['name'],
         fn_input_types=','.join(
-            [f"{collapse_if_tuple(abi_input)}{' indexed' if abi_input['indexed'] else ''}" 
-            for abi_input in abi.get('inputs', [])]
+            [f"{collapse_if_tuple(abi_input)}{' indexed' if abi_input['indexed'] else ''}"
+                for abi_input in abi.get('inputs', [])]
         ),
     )
+
+
+def retrieve_stats_from_import_results(raw_import_results):
+    num_processed = len(raw_import_results)
+
+    import_results = [
+        result
+        for result in raw_import_results
+        if result is not None
+    ]
+
+    num_ignored = num_processed - len(import_results)
+
+    if len(import_results) == 0:
+        num_imported = 0
+        num_duplicates = 0
+    else:
+        num_imported = sum(tuple(zip(*import_results))[1])
+        num_duplicates = len(import_results) - num_imported
+
+    ImportStats = namedtuple('ImportStats', ['num_processed', 'num_imported',
```

Comment:
You should define this at the top level of the module instead of inside this function context so that you don't incur the extra cost of class creation at runtime.